src/components/JsonEditorContainer/JsonEditorContainer.tsx

```diff
@@ -1,13 +1,18 @@
 import React, { useEffect, useRef } from "react";
-import JSONEditor, { EditableNode, JSONEditorOptions, JSONPath, MenuItem, MenuItemNode, Template } from "jsoneditor";
+import JSONEditor, { JSONEditorOptions, JSONPath, MenuItem, MenuItemNode, Template } from "jsoneditor";
 import "./JsonEditorContainer.scss";
 import { addNode, removeNode } from '../../validator/Validator';
 import { ErrorType } from "../../validator/interfaces/IValidationResult";
 import { StringNode } from "../../validator/nodes/StringNode";
 import { ArrayNode } from "../../validator/nodes/ArrayNode";
+import { IData } from "../../validator/nodes/BaseNode";
+import { AdditionalNode } from "../../validator/nodes/AdditionalNode";
 
-const createValidInsertMenu = (submenu: MenuItem[] | undefined, validInsertItems: any, existingKeys: (number | string)[]) => {
+const createValidInsertMenu = (submenu: MenuItem[] | undefined, currentJson: any, parentPath: (string | number)[]) => {
     const validMenuItems: MenuItem[] = [];
+    const validInsertItems: any = getValidInsertItems(parentPath);
+    const existingKeys: (number | string)[] = getExistingKeys(currentJson, [...parentPath]);
```

Comment:
in getExistingKeys method (line: 56) can we just return Object.keys(subTree)  instead of applying map operation.